plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -698,11 +698,16 @@ public void finishInsertTable(ConnectorSession session, JdbcOutputTableHandle ha
         finally {
             JdbcTableHandle temporaryTableHandle = new JdbcTableHandle(
                     new SchemaTableName(handle.getSchemaName(), handle.getTemporaryTableName()),
-                    new RemoteTableName(Optional.ofNullable(handle.getCatalogName()), Optional.ofNullable(handle.getSchemaName()), handle.getTemporaryTableName()));
+                    temporaryTable);
             dropTable(session, temporaryTableHandle);
         }
     }
 
+    protected String buildCopyTableSql(ConnectorSession session, RemoteTableName sourceTable, RemoteTableName targetTable, String columnNames)
```

Comment:
`columnNames` can be expected to be a List of actual column names
here it's a pre-quoted and `,`-concatenated

also, it's quoted while others are not
let's pass `handle.getColumnNames()` here, or whole `handle`

Reply:
good catch

Reply:
buildInsertSql?

Reply:
`buildInsertSql` is occupied by sql for inserts. This one is for copying data.
I'd happily use some widely used shortcut for `INSERT INTO SELECT FROM` if only it existed as one for `create table as select` exists: `CTAS`.

`buildInsertSql` cannot be reused because it's `INSERT INTO VALUES` and it's used for batch inserts.

Reply:
current `buildInsertSql` has different parameters

Reply:
Seems like 
```
buildInsertSql(session, targetTable, temporaryTable, columnNames)
```
perfectly matches logic for copying table.